odoo/fields.py

```diff
@@ -1402,6 +1423,40 @@ def determine_domain(self, records, operator, value):
         """ Return a domain representing a condition on ``self``. """
         return determine(self.search, records, operator, value)
 
+    # cache operations
+    def _cache(self, env):
+        # return logical readonly cache dict
+        return env.cache._get_field_cache(self, env.cache_key(self))
+
+    def set_cache(self, record, value, dirty=False, check_dirty=True):
+        # when storing the value of a content dependent(binary) field, it will be stored once under the context value
+        # `context.get('bin_size')`, and a second time under the context value `None`. The flush implementation will
+        # then retrieve the value using the context value `None`.
+        env = record.env
+        context_key = env.cache_key(self)
+        env.cache.set(self, context_key, record._ids[0], value, dirty=dirty, check_dirty=check_dirty)
+        if check_dirty and dirty and context_key is not None:
+            # put the values under conventional context key values None,
+            # in order to ease the retrieval of those values to flush them
+            env.cache.set(self, None, record._ids[0], value, dirty=False, check_dirty=False)
+
+    def update_cache(self, records, values, dirty=False, check_dirty=True):
+        env = records.env
+        context_key = env.cache_key(self)
+        env.cache.update(self, context_key, records._ids, values, dirty=dirty, check_dirty=check_dirty)
+        if check_dirty and dirty and context_key is not None:
+            # put the values under conventional context key values None,
+            # in order to ease the retrieval of those values to flush them
+            env.cache.update(self, None, records._ids, values, dirty=False, check_dirty=False)
+
+    def insert_cache_missing(self, records, values):
```

Comment:
I would name it `insert_cache`, and make it clear in its (missing) docstring that it does not modify existing cache entries.